lib/client.js

```diff
@@ -212,16 +212,20 @@ Client.prototype._invoke = function(method, args, location, callback, options, e
   self.lastRequest = xml;
 
   self.emit('message', message);
-  self.emit('request', xml);
+  self.emit('request', xml, name);
 
   req = self.httpClient.request(location, xml, function(err, response, body) {
     var result;
     var obj;
     self.lastResponse = body;
     self.lastResponseHeaders = response && response.headers;
-    self.emit('response', body);
+    self.emit('response', body, name);
```

Comment:
Adding a new event `soapSuccess`? We could. Then we only need to handle 2 events. But I assume we don't want to remove any existing events for backward compatibility.